edc-dataplane/edc-dataplane-base/build.gradle.kts

```diff
@@ -15,7 +15,8 @@ dependencies {
     implementation(edc.core.connector)
     implementation(edc.boot)
 
-
     implementation(edc.bundles.monitoring)
     implementation(edc.ext.http)
+
+    runtimeOnly(project(":edc-extensions:observability-api-customization"))
```

Comment:
our custom extension is a drop-in replacement, so `implementation(edc.api.observability)` (couple of lines above) should be deleted. And while you're there: could you replace all `implementation` dependencies with `runtimeOnly` dependencies?

also: nit - project-dependencies should be the first in the list.